File: 백준/Silver/2164. 카드2/카드2.c
#include<stdio.h>

int main(void)
{
	int N;

	scanf("%d", &N);

	if(N<3)
		printf("%d", N);
	else
	{
		int index=0;

		while((1<<++index)<N);
		N-=1<<index-1;

		printf("%d", N<<1);
	}

	return 0;
}

File: 백준/Silver/2164. 카드2/카드2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));

		int N=Integer.parseInt(br.readLine());
		
		if(N<3)
			System.out.println(N);
		else {
			int index=0;
			
			while(1<<++index<N);
			N-=1<<index-1;
			
			System.out.println(N<<1);
		}
	}
}